lua/precognition/init.lua

```diff
@@ -35,7 +35,7 @@ local M = {}
 ---@class Precognition.PartialConfig
 ---@field startVisible? boolean
 ---@field showBlankVirtLine? boolean
----@field highlightColor? string
+---@field highlightColor? string | vim.api.keyset.highlight
```

Comment:
I'm not sure, my Lua experience is very minimal, but it was shown in the LSP doc help. Here's the copy from goto-definition source:

```
--- @param val vim.api.keyset.highlight Highlight definition map, accepts the following keys:
---            • fg: color name or "#RRGGBB", see note.
---            • bg: color name or "#RRGGBB", see note.
---            • sp: color name or "#RRGGBB"
---            • blend: integer between 0 and 100
---            • bold: boolean
---            • standout: boolean
---            • underline: boolean
---            • undercurl: boolean
---            • underdouble: boolean
---            • underdotted: boolean
---            • underdashed: boolean
---            • strikethrough: boolean
---            • italic: boolean
---            • reverse: boolean
---            • nocombine: boolean
---            • link: name of another highlight group to link to, see
---              `:hi-link`.
---            • default: Don't override existing definition `:hi-default`
---            • ctermfg: Sets foreground of cterm color `ctermfg`
---            • ctermbg: Sets background of cterm color `ctermbg`
---            • cterm: cterm attribute map, like `highlight-args`. If not
---              set, cterm attributes will match those from the attribute map
---              documented above.
---            • force: if true force update the highlight group when it
---              exists.
function vim.api.nvim_set_hl(ns_id, name, val) end
```

Alternatives are just use `table`, or make a custom class like you've done for the other options. Was hoping to avoid redefining a type if possible, but it also depends on whether you want to accept the full range of highlight options.